runtime/syntax/dart.vim

```diff
@@ -0,0 +1,89 @@
+" Vim syntax file
+"
+" Language:     Dart
+" Maintainer:   Eugene 'pr3d4t0r' Ciurana <dart.syntax AT cime.net >
+" URL:          https://github.com/pr3d4t0r/dart-vim-syntax
+" Last Change:  See GitHub URL
+"
+" vim: set fileencoding=utf-8:
+
+
+" Support links (remove in a future version of this syntax file):
+"
+" https://github.com/Houl/ExplainPattern-vim
+" http://vimregex.com/
+" http://vimhelp.appspot.com/pattern.txt.html#pattern%2Etxt
+
+
+" Quit when a (custom) syntax file was already loaded
+if exists("b:current_syntax")
+  finish
+endif
+
+let s:cpo_save = &cpo
+set cpo&vim
+
+
+syn keyword dartCommentTodo     contained TODO FIXME XXX TBD
+syn match   dartLineComment     "//.*" contains=dartTodo,@Spell
+syn match   dartCommentSkip     "^[ \t]*\*\($\|[ \t]\+\)"
+syn region  dartComment         start="/\*"  end="\*/" contains=@Spell,dartTodo
+syn keyword dartReserved        assert async await class const export extends external final hide import implements interface library mixin on show super sync yield
+syn match   dartNumber          "-\=\<\d\+L\=\>\|0[xX][0-9a-fA-F]\+\>"
```

Comment:
You could replace [0-9a-fA-Z] with \x.
Matching can then be faster when using "set regexpengine=1" (old regex engine).